testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -674,10 +674,10 @@ def pillar_get(key, minion)
   node = get_target(host)
   if host.include? 'ceos'
     node.run('yum -y remove --setopt=clean_requirements_on_remove=1 salt salt-minion', false)
-  elsif host.include? 'ubuntu'
+  elsif host.include? 'ubuntu' or host.include? 'debian'
     node.run('apt-get --assume-yes remove salt-common salt-minion && apt-get purge salt-common salt-minion && apt-get autoremove', false)
   else
-    node.run('zypper --non-interactive remove --clean-deps -y salt salt-minion', false)
+    node.run('zypper --non-interactive remove --clean-deps -y salt salt-minion spacewalk-proxy-salt', false)
```

Comment:
This should only run in proxy right? Otherwise does it throw an error or a different exit code than expected? Since we run this step to all minions

Reply:
It indeed runs everywhere, but it does not throw an error if one of the packages is missing - the `false` argument makes it ignore errors.

In practice, error 140 appears when some packages are removed and others not, and it is then ignored.

Reply:
all good then